### PR TITLE
roachtest: add uniform distribution ycsb

### PR DESCRIPTION
**Pull request by DarrylWong**
This change adds roachtests for ycsb workloads using uniform distribution. Currently we only test with zipfian distribution. While zipfian is a more realistic distribution, testing uniform can still serve as a baseline/canary to compare against.

Release note: none
Epic: none
Fixes: none

### CODE REVIEW

**Comment by DarrylWong** [comment]
GCE Runs: [A-E](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14666417?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildTestsSection=true), [F](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14658197?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true)

`Zipfian` data was pulled from [this nightly run](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14643937?buildTab=tests&name=ycsb).

**Throughputs** **(zipfian first, uniform second):**

Workload A:
image

image


Workload B:
image

image



Workload C:
(was preempted, used[ this run instead](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14635121?buildTab=tests&name=ycsb&expandedTest=build%3A%28id%3A14635121%29%2Cid%3A13833&showLog=14635121_13764_2388&logFilter=debug&logView=flowAware)) 
image

image


Workload D:
image

image


Workload E:
image

image



Workload F:
image


image

---------

Across the board we see `uniform` distribution do the same or better, as expected. The more interesting workloads are probably the ones where `uniform` outperforms `zipfian`, workloads A and F. This also is expected as those two workloads are the most write heavy and have the most contention, where the hotspots created by `zipfian` would slow things down more.

File: pkg/cmd/roachtest/tests/ycsb.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ func registerYCSB(r registry.Registry) {
 	cpusConfigs := []int{8, 32}
 	cpusWithReadCommitted := 32
 	cpusWithGlobalMVCCRangeTombstone := 32
+	cpusWithUniformDistribution := 32
 
 	// concurrencyConfigs contains near-optimal concurrency levels for each
 	// (workload, cpu count) combination. All of these figures were tuned on GCP
```
Comment (renatolabs) [inline pkg/cmd/roachtest/tests/ycsb.go:38]:
```go
// All of these figures were tuned on GCP n1-standard instance types.
```

🤔 

(nothing for you to do here, but these numbers might benefit from another look at some point.)

**Review by renatolabs — APPROVED** [review]
Nice!

File: pkg/cmd/roachtest/tests/ycsb.go (hunk 1 above)
Comment (DarrylWong) [reply pkg/cmd/roachtest/tests/ycsb.go:38]: The cpu usage looked fairly saturated for most of the runs I looked at, but yeah it wouldn't hurt to fine tune them.

**Comment by DarrylWong** [comment]
TFTR!

bors r=renatolabs

**Pull request merged**